Libs/MRML/Core/vtkMRMLModelDisplayNode.h

```diff
@@ -73,23 +90,48 @@ class VTK_MRML_EXPORT vtkMRMLModelDisplayNode : public vtkMRMLDisplayNode
   /// \sa SetActiveScalarName()
   virtual void SetActiveAttributeLocation(int location);
 
-  /// Update the pipeline based on this node attributes
-  virtual void UpdatePolyDataPipeline();
+  /// Reimplemented to update scalar range accordingly
+  /// \sa SetActiveScalarName()
+  virtual void SetScalarRangeFlag(int flag);
+
+  /// Set whether to threshold the model display node.
+  /// \sa ThresholdEnabled, GetThresholdEnabled()
+  vtkSetMacro(ThresholdEnabled,bool);
+  vtkBooleanMacro(ThresholdEnabled,bool);
+
+  /// Get whether to threshold the model display node.
+  /// \sa ThresholdEnabled, SetThresholdEnabled(), ThresholdEnabledOn(),
+  /// ThresholdEnabledOff()
+  vtkGetMacro(ThresholdEnabled,bool);
+
+  /// Set the threshold range of the model display node.
+  /// \sa GetThresholdMin(), GetTresholdMax()
+  void SetThresholdRange(double min, double max);
+  void SetThresholdRange(double range[2]);
+
+  /// Get the threshold range of the model display node.
+  /// \sa SetThresholdRange()
+  void GetThresholdRange(double range[2]);
+  double GetThresholdMin();
+  double GetThresholdMax();
 
 protected:
   vtkMRMLModelDisplayNode();
   ~vtkMRMLModelDisplayNode();
   vtkMRMLModelDisplayNode(const vtkMRMLModelDisplayNode&);
   void operator=(const vtkMRMLModelDisplayNode&);
 
-  virtual void ProcessMRMLEvents(vtkObject *caller,
```

Comment:
> I've added a breakpoint into vtkMRMLModelDisplayNode::ProcessMRMLEvents and I don't see any unreasonable ModifiedEvent calls. Could you check in your stack trace where those hundreds of ModifiedEvent calls originated from?

Ok I figured out what happened: I was running the model tests and the `EXERCISE_ALL_BASIC_MRML_METHODS` as well as other subsets preceding the test I was debugging would just keep triggering `ProcessMRMLEvents`, for a good reason I guess.

However, three things happen in `UpdatePolyDataPipeline()` (now `UpdateAssignedAttribute()`)
1. reassign the scalar name and location in the `AssignAttribute` filter
2. Update the `AssignAttribute` filter
3. if scalar range mode is not manual, update scalar range based on data, colornode, or data type (within `UpdateScalarRange()`)

Updating the Assignattribute filter should only happen when ActiveScalarName or ActiveScalarLocation are updated. Would it make sense to call `UpdateScalarRange()` only in `ProcessMRMLEvents()` ?